importlib_metadata/__init__.py

```diff
@@ -562,15 +562,31 @@ def _read_files_egginfo_installed(self):
         if not text or not subdir:
             return
 
+        site_path = self.locate_file('').resolve()
         paths = (
-            (subdir / name)
-            .resolve()
-            .relative_to(self.locate_file('').resolve())
-            .as_posix()
+            self._relative_to(
+                (subdir / name).resolve(),
+                site_path,
+            ).as_posix()
             for name in text.splitlines()
         )
         return map('"{}"'.format, paths)
 
+    def _relative_to(self, path, root):
+        """
+        Workaround for https://github.com/python/cpython/issues/67271 where ".."
+        isn't added by pathlib.Path.relative_to() when path is not
+        a subpath of root.
+
+        One example of such a package is dask-labextension, which uses
+        jupyter-packaging to install JupyterLab javascript files outside
+        of site-packages.
+        """
+        try:
+            return path.relative_to(root)
+        except ValueError:
+            return pathlib.Path(os.path.relpath(path, root))
+
```

Comment:
It looks like the bug was fixed in Python 3.9. We should move this workaround to .compat/py38 so it can be removed when no longer needed and can be omitted when porting this code into CPython.

Reply:
Oh, so I tried extracting this function into a wrapper that's only applied on Python 3.9 and later, but then the ValueError is emitted. So now I no longer trust the comment, which indicates that this is a workaround for a bug that's closed.

Reply:
Aha. So I found that there is https://github.com/python/cpython/issues/84538, which captures the underlying issue, and the fix for which adds a `walk_up` parameter.

Reply:
Fixed in 1584b96. Now the diff for Python 3.12+ is simply to use `walk_up=True`.